libs/api/domains/driving-license-book/src/drivingLicenseBook.resolver.ts

```diff
@@ -0,0 +1,71 @@
+import { Args, Mutation, Query, Resolver } from '@nestjs/graphql'
+import { UseGuards } from '@nestjs/common'
+import {
+  CurrentUser,
+  IdsUserGuard,
+  ScopesGuard,
+} from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+import { DrivingLicenseBookService } from './drivingLicenseBook.service'
+import { StudentListInput } from './dto/studentList.input'
+import { DrivingBookStudent } from './models/drivingBookStudent.response'
+import { PracticalDrivingLessonsInput } from './dto/getPracticalDrivingLessons.input'
+import { PracticalDrivingLesson } from './models/practicalDrivingLesson.response'
+import { StudentInput } from './dto/student.input'
+import { DrivingBookStudentOverview } from './models/drivingBookStudentOverview.response'
+import { CreatePracticalDrivingLessonInput } from './dto/createPracticalDrivingLesson.input'
+import { SuccessResponse } from './models/success.response'
+import { UpdatePracticalDrivingLessonInput } from './dto/updatePracticalDrivingLesson.input'
+import { DeletePracticalDrivingLessonInput } from './dto/deletePracticalDrivingLesson.input'
+import { Student } from './models/studentsTeacherNationalId.response'
+
+@UseGuards(IdsUserGuard, ScopesGuard)
+@Resolver()
+export class DrivinLicenseBookResolver {
+  constructor(
+    private readonly drivingLicenseBookService: DrivingLicenseBookService,
+  ) {}
+
+  @Query(() => [DrivingBookStudent], { nullable: true })
+  drivingBookStudentList(@Args('input') input: StudentListInput) {
+    return this.drivingLicenseBookService.getStudentList(input)
+  }
+
+  @Query(() => [Student])
+  drivingBookStudentListByTeacherNationalId(@CurrentUser() user: User) {
```

Comment:
This comment depends on if the query above should be open or not.
`ByTeacherNationalId` this should most likely be a part of the input ... or if the previous query gets removed.. than this name should just be:
```typescript
drivingLicenseBookStudents
```

Reply:
Or actually .. should be named `drivingLicenseBookStudents` ... but the input should contain a boolean specifying something like `fetchMyStudents: true`